src/Migration/Action_Migrator.php

```diff
@@ -38,6 +44,15 @@ public function migrate( $source_action_id ) {
 
 		try {
 			$destination_action_id = $this->destination->save_action( $action );
+
+			// If the action is completed, make sure to set the dates properly.
+			if ( $action->is_finished() && $this->destination instanceof DB_Store && $this->source instanceof \ActionScheduler_wpPostStore ) {
+				$post = get_post( $source_action_id );
```

Comment:
There is no guarantee the old actions are stored as posts. They could come from any child of `ActionScheduler_Store`, so it's not safe to use `get_post()` here. Instead, `$this->source` needs to be used.

The obvious issue with that is that neither `ActionScheduler_wpPostStore` nor any other `ActionScheduler_Store` class currently has a way to access the "last attempt" time. That can be solved by adding a new abstract `get_last_attempt()` method to `ActionScheduler_Store` and then implementing that in `ActionScheduler_wpPostStore` (as well as `Hybrid_Store` and `DB_Store`).

One upside to that headache is that the method can return a `DateTime` instance, so there will be no need to mess about with different strings representing different timezones for that value (the `as_get_datetime_object()` function can be used for instantiating the `DateTime`).